src/CardsDeck/Cards.cpp

```diff
@@ -1,6 +1,248 @@
-// Deck -- > draw() 
+#include <algorithm>
+#include <iostream>
+#include <random>
+#include "Cards.h"
+#include "Orders.h"
 
+/************************************************************ Card ************************************************************/
+const vector<string> Card::cardTypes = {"Bomb", "Reinforcement", "Blockade", "Airlift", "Diplomacy"}; // vector holding all card types
 
-// Hand
+// default constructor
+Card::Card() {
+    type = nullptr;
+}
 
-// play()
\ No newline at end of file
+// destructor
+Card::~Card() {
+    delete type;
+}
+
+// parametrized constructor
+Card::Card(const string &cardType) {
+    type = nullptr;
+    if (isValidCardType(cardType)) {
+        type = new string(cardType);
+    } else {
+        cout << "Invalid card type: " << cardType << endl;
+    }
+}
+
+// deep copy constructor
+Card::Card(const Card &other) {
+    type = nullptr;
+    if (other.type != nullptr) {
+        type = new string(*(other.type));
+    }
+}
+
+// assignment operator
+Card &Card::operator=(const Card &other) {
+    if (this != &other) {
+        delete type;
+        type = nullptr;
+        if (other.type != nullptr) {
+            type = new string(*(other.type));
+        }
+    }
+    return *this;
+}
+
+bool Card::isValidCardType(const string &value) {
+    for (size_t i = 0; i < cardTypes.size(); i++) {
+        if (value == cardTypes[i]) {
+            return true; // returns true if card is valid type
```

Comment:
you can just say `return value == cardTypes[i]`
it will return the actual value of this expression which can be true or false